rust_src/remacs-lib/docfile.rs

```diff
@@ -137,6 +138,24 @@ pub extern "C" fn scan_rust_file(
             let name = CString::new(&caps[1]).unwrap();
             let value = CString::new(&caps[2]).unwrap();
             add_global(SYMBOL, name.as_ptr(), 0, value.as_ptr());
+        } else if line.starts_with("defvar_") {
+            // defvar_lisp!(f_Vpost_self_insert_hook, "post-self-insert-hook", Qnil);
+            lazy_static! {
+                static ref RE: Regex = Regex::new(r#"defvar_(lisp)!\((.+?),\s+"(.+?)",\s+(.+?)\);"#).unwrap();
+            }
+            let caps = RE.captures(line).unwrap();
+            let kindstr = &caps[1];
+            let kind = match kindstr {
+                "lisp" => LISP_OBJECT,
+                "lisp_nopro" => LISP_OBJECT,
+                "bool" => BOOLEAN,
+                "int" => EMACS_INTEGER,
+                _ => panic!("oops"),
+            };
+            let fvname = &caps[2];
+            assert!(fvname.starts_with("f_V"));
+            let vname = CString::new(fvname.splitn(2, "_").nth(1).unwrap()).unwrap();
```

Comment:
Any chance we could move to using the '?' operator here?

`let vname = CString::new(fvname.splitn(2, "_").nth(1)?)?;`

Reply:
Then it would have to return a `Result`.

Reply:
Right. I wonder if that might be better long term for this particular code.

Reply:
It might be! I'll think about preparing a second pull request with that kind of thing in it.